services/121-service/src/queues-registry/register-processor.decorator.ts

```diff
@@ -0,0 +1,21 @@
+import { Processor } from '@nestjs/bull';
+
+const ProcessorRegistry = new Set<string>();
+
+export function RegisteredProcessor(
+  queueName: string,
+  scope?: any,
+): ClassDecorator {
+  return (target) => {
```

Comment:
Maybe also check that the processor name has not been added yet to prevent duplicates?